src/shared/Core/ApplicationBase.cs

```diff
@@ -75,6 +77,31 @@ public Task<int> RunAsync(string[] args)
                 Context.Trace.WriteLine("Tracing of secrets is enabled. Trace output may contain sensitive information.");
             }
 
+            // Enable TRACE2 tracing
+            if (Context.Settings.GetTrace2TracingEnabled(out string trace2Value))
+            {
+                var connectionPath = PlatformUtils.IsWindows() ? trace2Value.TrimUntilLastIndexOf("/") :
+                    trace2Value.TrimUntilIndexOf(":");
+
+                if (!string.IsNullOrEmpty(connectionPath))
+                {
+                    Context.Trace2.AddListener(new CollectorWriter(
+                            () => new NamedPipeClientStream(".", connectionPath, PipeDirection.Out,
+                                PipeOptions.Asynchronous)
+                        )
+                    );
+
+                    if (AssemblyUtils.TryGetAssemblyVersion(out string version))
+                    {
+                        Context.Trace2.WriteVersion(version);
+                    }
+
+                    var argv = new List<string>() { Context.ApplicationPath };
+                    argv.AddRange(args);
+                    Context.Trace2.WriteStart(argv, Context.ApplicationTimer.ElapsedMilliseconds);
```

Comment:
I assume the application timer get used to timestamp all events based on the current `ElapsedMilliseconds` since app start?

Why don't we just store/accept a `DateTime` (or `DateTimeOffset`, or UNIX 64-bit timestamp `long`) that is computed once in `Main` and passed through to `CommandContext` and `Trace2`?

Elapsed times can be computed instead of maintaining a running `Stopwatch`.

Reply:
Yeah, just get the current utc time in Main() and let each event generator do a Now() to emit in the record and let that routine do the subtraction to get the `t_abs` field for the record. (Yes, this is a redundancy in the format.)

Later, we'll need to talk about region-relative stuff and they have a hierarchy of clocks for nested regions.  I can explain later.

Reply:
Ignore my previous question on this topic - thanks for answering here!